src/transformers/pipelines.py

```diff
@@ -553,6 +553,232 @@ def __call__(self, *args, **kwargs):
+    which includes the uni-directional models in the library (e.g. gpt2).
+    See the list of available community models on
+    `huggingface.co/models <https://huggingface.co/models?search=&filter=lm-head>`__.
+
+    Arguments:
+        model (:obj:`~transformers.PreTrainedModel` or :obj:`~transformers.TFPreTrainedModel`):
+            The model that will be used by the pipeline to make predictions. This needs to be a model inheriting from
+            :class:`~transformers.PreTrainedModel` for PyTorch and :class:`~transformers.TFPreTrainedModel` for
+            TensorFlow.
+        tokenizer (:obj:`~transformers.PreTrainedTokenizer`):
+            The tokenizer that will be used by the pipeline to encode data for the model. This object inherits from
+            :class:`~transformers.PreTrainedTokenizer`.
+        modelcard (:obj:`str` or :class:`~transformers.ModelCard`, `optional`, defaults to :obj:`None`):
+            Model card attributed to the model for this pipeline.
+        framework (:obj:`str`, `optional`, defaults to :obj:`None`):
+            The framework to use, either "pt" for PyTorch or "tf" for TensorFlow. The specified framework must be
+            installed.
+
+            If no framework is specified, will default to the one currently installed. If no framework is specified
+            and both frameworks are installed, will default to PyTorch.
+        args_parser (:class:`~transformers.pipelines.ArgumentHandler`, `optional`, defaults to :obj:`None`):
+            Reference to the object in charge of parsing supplied pipeline parameters.
+        device (:obj:`int`, `optional`, defaults to :obj:`-1`):
+            Device ordinal for CPU/GPU supports. Setting this to -1 will leverage CPU, >=0 will run the model
+            on the associated CUDA device id.
+    """
+
+    # Padding text to help Transformer-XL and XLNet with short prompts as proposed by Aman Rusia
+    # in https://github.com/rusiaaman/XLNet-gen#methodology
+    # and https://medium.com/@amanrusia/xlnet-speaks-comparison-to-gpt-2-ea1a4e9ba39e
+    PADDING_TEXT = """In 1991, the remains of Russian Tsar Nicholas II and his family
+    (except for Alexei and Maria) are discovered.
+    The voice of Nicholas's young son, Tsarevich Alexei Nikolaevich, narrates the
+    remainder of the story. 1883 Western Siberia,
+    a young Grigori Rasputin is asked by his father and a group of men to perform magic.
+    Rasputin has a vision and denounces one of the men as a horse thief. Although his
+    father initially slaps him for making such an accusation, Rasputin watches as the
+    man is chased outside and beaten. Twenty years later, Rasputin sees a vision of
+    the Virgin Mary, prompting him to become a priest. Rasputin quickly becomes famous,
+    with people, even a bishop, begging for his blessing. <eod> </s> <eos>"""
+
+    def __init__(
```

Comment:
We should not need a `__init__` function here at all.

1. Generation specific parameters, such as: 
`length`, `temperature`, `top_k`, `top_p`, ... will be set either through `task_specific_params` as is done in the Bart config e.g. for summarization:
https://s3.amazonaws.com/models.huggingface.co/bert/facebook/bart-large-cnn/config.json
(we will add a `generation` task specific dict for generation to all models that can generate)
or by using `generate_kwargs` for the generation method as done in the summarization pipeline:
https://github.com/huggingface/transformers/blob/01c37dcdb529bff38aadd51001cb5812e5fe9b21/src/transformers/pipelines.py#L1257

2. I think we will remove `xlm` from the language generation models for now - it does not work well at all.

3. the padding text will be set when doing "__call__"

Reply:
Is there also a need to include `inputs["attention_mask"]` as an argument to `generation`? Was under the impression that it's not necessary for doing the GPT-2 forward pass. Also it's not used in [run_generation.py](https://github.com/huggingface/transformers/blob/60a42ef1c04591e0709429276ccbc02608b7d47d/examples/run_generation.py#L224).

Reply:
Good question! It's currently needed for "summarization" and "translation", but not really needed for "generation" at the moment (It will be needed when batch generation is implemented).

Reply:
And we can now remove the whole `def __init__(...)` function since it's equivalent to the init of the class `Pipelines` from which `TextGeneration` inherets from. See https://github.com/huggingface/transformers/blob/a21d4fa410dc3b4c62f93aa0e6bbe4b75a101ee9/src/transformers/pipelines.py#L334